classes/Product.php

```diff
@@ -3752,7 +3752,9 @@ public static function getQuantity(
         $availableQuantity = StockAvailable::getQuantityAvailableByProduct($idProduct, $idProductAttribute);
         $nbProductInCart = 0;
 
-        if (!empty($cart)) {
+        // we don't substract products in cart if the cart is already attached to an order, since stock quantity
+        // has already been updated, this is only useful when the order has not yet been created
+        if (!empty($cart) && (int) $cart->id_order <= 0) {
```

Comment:
@matthieu-rolland as discussed based on the comment from @Matt75 https://github.com/PrestaShop/PrestaShop/pull/21795#issuecomment-739249918

It might not be a good idea to store the orderId in the cart table because it maybe be associated to multiple orders, even though in this case it wouldn't have much impact because we just need to know if the Cart is order, no matter how many orders a linked to it. So we have two solutions here I guess:
- never store the relation and rely on `Order::getIdByCartId` this will make an extra SQL query but at least we are sure to have up to date data and nothin in the cart table is confusing
- we store the Order `reference` instead of its ID, since all split orders have the same reference we can still correctly get the whole list thanks to this column

Reply:
So I opted for solution number 2